diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -374,9 +374,15 @@ namespace diff_drive_controller{
     config.publish_rate = publish_rate;
     config.enable_odom_tf = enable_odom_tf_;
 
-    dyn_reconf_server_ = std::make_shared<ReconfigureServer>(controller_nh);
+    dyn_reconf_server_ = boost::make_shared<ReconfigureServer>(mDynServerMutex, controller_nh);
```

Comment:
I am sorry for my second message, but my curiosity pushed me to compile again the code.

I am a little rusty, but I do not think that we can avoid to use boost here. The `dynamic_reconfigure::Server` use `boost::recursive_mutex` and I do not remember any trick to convert a `boost` mutex to `std`.

The reference is of `dynamic_reconfigure::Server` is here: https://github.com/ros/dynamic_reconfigure/blob/melodic-devel/include/dynamic_reconfigure/server.h#L75

Anyway I will sleep on it!
Tomorrow I will send you a commit with all reviewed parts.

Reply:
I haven't looked at the rest of this PR yet, but the `shared_ptr` can definitely still be `std`. Passing the `boost::recursive_mutex` as an argument to the constructor/`make_shared` doesn't affect the type of the `shared_ptr`.

But yes, we can't pass a `std::recursive_mutex` to that constructor since it asks for `boost`. We will have to live with that Boost usage.